Networking/Networking/ApplicationPassword/RequestProcessor.swift

```diff
@@ -0,0 +1,87 @@
+import Alamofire
+import Foundation
+
+/// Authenticates and retries requests
+///
+final class RequestProcessor {
+    private var requestsToRetry = [RequestRetryCompletion]()
+
+    private var isAuthenticating = false
+
+    private let requestAuthenticator: RequestAuthenticator
+
+    init(credentials: Credentials?) {
+        requestAuthenticator = RequestAuthenticator(credentials: credentials)
+    }
+}
+
+// MARK: Request Authentication
+//
+extension RequestProcessor: RequestAdapter {
+    func adapt(_ urlRequest: URLRequest) throws -> URLRequest {
+        return try requestAuthenticator.authenticate(urlRequest)
+    }
+}
+
+// MARK: Retrying Request
+//
+extension RequestProcessor: RequestRetrier {
+    func should(_ manager: Alamofire.SessionManager,
+                retry request: Alamofire.Request,
+                with error: Error,
+                completion: @escaping Alamofire.RequestRetryCompletion) {
+        guard
+            request.retryCount == 0, // Only retry once
+            let urlRequest = request.request,
+            requestAuthenticator.shouldRetry(urlRequest), // Retry only REST API requests that use application password
+            shouldRetry(error) // Retry only specific errors
+        else {
+            return completion(false, 0.0)
+        }
+
+        requestsToRetry.append(completion)
+        if !isAuthenticating {
+            generateApplicationPassword()
+        }
+    }
+}
+
+// MARK: Helpers
+//
+private extension RequestProcessor {
+    func generateApplicationPassword() {
+        Task(priority: .medium) {
+            isAuthenticating = true
+
+            do {
+                let _ = try await requestAuthenticator.generateApplicationPassword()
+                isAuthenticating = false
+                completeRequests(true)
+            } catch {
+                isAuthenticating = false
+                completeRequests(false)
+            }
+        }
+    }
+
+    func shouldRetry(_ error: Error) -> Bool {
+        // Need to generate application password
+        if .applicationPasswordNotAvailable == error as? RequestAuthenticatorError {
+            return true
+        }
+
+        // Failed authorization
+        if case .responseValidationFailed(reason: .unacceptableStatusCode(code: 401)) = error as? AFError {
+            return true
+        }
```

Comment:
might this trigger a retry for non-REST requests? in the issue description https://github.com/woocommerce/woocommerce-ios/issues/8482, it mentions there's no retry for WPCOM/Jetpack-tunneled requests so just making sure.

Reply:
[This](https://github.com/woocommerce/woocommerce-ios/pull/8483/files#diff-e2b63cf38f66c7a410720d0644c041faf8dbbca67d1e011e7ce325bd698a596dR36) will take care of filtering REST requests.